### ISSUE TITLE
Retry attempts are only done for different hostnames

### ISSUE THREAD

**Issue by benzman81** [aerokube/ggr#114]
I reduced the timeout to 10s an get the message "cannot create session xxx on any hosts after 1 attempt(s)" pretty often. I digged a bit into the code and was wondering why it is never more than 1 attempt.
The issue is the hostname. I have all my windows nodes running in a virtualbox on one linux host. I used the linuxhostname with the different ports with portforwarding first and that causes always only 1 attempt.
To solve this issue I needed to create a private network where each node gets its own ip. After that, multiple attempts will be done. The private network is a bit too much, so I would love a solution with same hostname as I think uniqueness should be with hostname AND port. Is that by design or am I doing something wrong?

**Comment by vania-pooh** [comment]
@benzman81 in fact we are using full hub URL instead of hostname to distribute the load. Take a look at how we calculate distinct host hash sum when placing it into the hash map: https://github.com/aerokube/ggr/blob/master/config.go#L84-L98

**Comment by benzman81** [comment]
@vania-pooh yes, I saw that, but in https://github.com/aerokube/ggr/blob/269fcdd0883714a882c824385a8a26943c7347ca/proxy.go#L226 the loop only uses hosts to make attempts and so only does one attempt.

**Comment by aandryashin** [comment]
Could you please provide us with your browsers.xml configuration?

8 сент. 2017 г. 11:20 ДП пользователь "Markus Krüger" <
notifications@github.com> написал:

**Comment by benzman81** [comment]
sure, here it is:

```
<qa:browsers xmlns:qa="urn:config.gridrouter.qatools.ru">
<browser name="internet explorer" defaultVersion="11">
    <version number="11">
        <region name="1">
            <host name="selenium-grid-ecmrd01" port="4450" count="1"/>
            <host name="selenium-grid-ecmrd01" port="4451" count="1"/>
            <host name="selenium-grid-ecmrd01" port="4452" count="1"/>
            <host name="selenium-grid-ecmrd01" port="4453" count="1"/>
            <host name="selenium-grid-ecmrd01" port="4454" count="1"/>
            <host name="selenium-grid-ecmrd01" port="4455" count="1"/>
            <host name="selenium-grid-ecmrd01" port="4456" count="1"/>
            <host name="selenium-grid-ecmrd01" port="4457" count="1"/>
            <host name="selenium-grid-ecmrd01" port="4458" count="1"/>
            <host name="selenium-grid-ecmrd01" port="4459" count="1"/>
        </region>
    </version>
</browser>
</qa:browsers>
```

**Comment by benzman81** [comment]
I also tried to create 10 regions with each one host. didnt help.

**Comment by aandryashin** [comment]
Thank you. We will try to reproduce the issue and update you soon.

8 сент. 2017 г. 12:14 ПП пользователь "Markus Krüger" <
notifications@github.com> написал:

**Comment by benzman81** [comment]
Great, thx for the quick help.

**Comment by aandryashin** [comment]
Thank you for reporting this issue, I was able to reproduce and fix:

https://github.com/aerokube/ggr/pull/115


2017-09-08 12:21 GMT+03:00 Markus Krüger <notifications@github.com>:

**Comment by vania-pooh** [comment]
@benzman81 released this fix as [1.2.7](https://github.com/aerokube/ggr/releases/tag/1.2.7).

**Comment by benzman81** [comment]
verified, thx
